src/calibrationlib/operator_feature_export.cpp

```diff
@@ -0,0 +1,107 @@
+#include <regex>
+
+#include <expression/expression_utils.hpp>
+#include <expression/lqp_column_expression.hpp>
+#include <logical_query_plan/stored_table_node.hpp>
+#include <utils/assert.hpp>
+#include "csv_writer.hpp"
+#include "hyrise.hpp"
+#include "operator_feature_export.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+OperatorFeatureExport::OperatorFeatureExport(const std::string& path_to_dir) : _path_to_dir(path_to_dir) {}
+
+void OperatorFeatureExport::export_to_csv(std::shared_ptr<const AbstractOperator> op) const {
+  if (op != nullptr) {
+    // Export current operator
+    _export_typed_operator(op);
+
+    export_to_csv(op->input_left());
+    export_to_csv(op->input_right());
+  }
+}
+
+void OperatorFeatureExport::_export_typed_operator(std::shared_ptr<const AbstractOperator> op) const {
+  switch (op->type()) {
+    case OperatorType::TableScan:
+      _export_table_scan(op);
+      break;
+    default:
+      break;
+  }
+}
+
+// Export features of a table scan operator
+void OperatorFeatureExport::_export_table_scan(std::shared_ptr<const AbstractOperator> op) const {
+  DebugAssert(op->type() == OperatorType::TableScan, "Expected operator of type: TableScan but got another one");
+
+  auto csv_writer = _csv_writers.at(op->type());
```

Comment:
Since this method might be executed quite often, performance is important. We could use `[]` without bounds checking and use a DebugAssert to be safe.